hugegraph-api/src/main/java/com/baidu/hugegraph/api/graph/UpdateStrategy.java

```diff
@@ -0,0 +1,176 @@
+            return combineSet(oldProperty, newProperty, INTERSECTION);
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            UNION.checkPropertyType(oldProperty, newProperty);
+        }
+    },
+
+    // Batch update Set should use union because of higher efficiency
+    APPEND {
+        @Override
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            ((Collection) oldProperty).addAll((Collection) newProperty);
+            return oldProperty;
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            UNION.checkPropertyType(oldProperty, newProperty);
+        }
+    },
+
+    ELIMINATE {
+        @Override
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            ((Collection) oldProperty).removeAll((Collection) newProperty);
+            return oldProperty;
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            UNION.checkPropertyType(oldProperty, newProperty);
+        }
+    };
+
+    abstract Object updatePropertyValue(Object oldProperty, Object newProperty);
+
+    abstract void checkPropertyType(Object oldProperty, Object newProperty);
+
+    public Object checkAndUpdateProperty(Object oldProperty,
+                                         Object newProperty) {
+        checkPropertyType(oldProperty, newProperty);
+        return updatePropertyValue(oldProperty, newProperty);
+    }
+
+    private static String formatError(Object oldProperty, Object newProperty,
+                                      String className) {
+        return String.format("Property must be %s, but got %s, %s", className,
+                             oldProperty.getClass().getSimpleName(),
+                             newProperty.getClass().getSimpleName());
+    }
+
+    private static Object compareNumber(Object oldProperty, Object newProperty,
+                                        UpdateStrategy strategy) {
+        Number oldNum = NumericUtil.convertToNumber(oldProperty);
+        Number newNum = NumericUtil.convertToNumber(newProperty);
+        int result = NumericUtil.compareNumber(oldNum, newNum);
+        return strategy == BIGGER ? (result > 0 ? oldProperty : newProperty) :
+                                    (result < 0 ? oldProperty : newProperty);
+    }
+
+    private static Set<?> combineSet(Object oldProperty, Object newProperty,
+                                     UpdateStrategy strategy) {
+        Set oldSet = oldProperty instanceof Set ?
+                     (Set) oldProperty : new HashSet((List) oldProperty);
+        Set newSet = newProperty instanceof Set ?
+                     (Set) newProperty : new HashSet((List) newProperty);
+        return strategy == UNION ? Sets.union(oldSet, newSet) :
+                                   Sets.intersection(oldSet, newSet);
```

Comment:
Set is a raw type. References to generic type Set<E> should be parameterized.

please update to:
```
        Set<?> oldSet = oldProperty instanceof Set ?
                        (Set<?>) oldProperty : 
                        new HashSet<>((List<?>) oldProperty);
        Set<?> newSet = newProperty instanceof Set ?
                        (Set<?>) newProperty : 
                        new HashSet<>((List<?>) newProperty);
        return strategy == UNION ? Sets.union(oldSet, newSet) :
               Sets.intersection(oldSet, newSet);
```